airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/MySqlStreamingQueryConfig.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.source.mysql;
+
+import io.airbyte.db.jdbc.streaming.AdaptiveStreamingQueryConfig;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.Statement;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MySqlStreamingQueryConfig extends AdaptiveStreamingQueryConfig {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(MySqlStreamingQueryConfig.class);
+
+  public MySqlStreamingQueryConfig() {
+    super();
+  }
+
+  @Override
+  public void initialize(final Connection connection, final Statement preparedStatement) throws SQLException {
+    connection.setAutoCommit(false);
+    preparedStatement.setFetchSize(Integer.MIN_VALUE);
```

Comment:
Can you explain this? 
Why is changing fetch size from 10 rows needed?
What is the meaning of putting a negative value (-2^31)

Reply:
More in the [ticket](https://github.com/airbytehq/airbyte/issues/8890) but in short: 
We have an issue to fetch data when mysql db table has more than 1 000 000 records (approximately).  Because we are using `jdbcUrl.append("?useCursorFetch=true")` and fetch size 10 it loads MySql resources significantly (disk storage and memory). So if instance has enough resources it will lead to a long waiting before the sync start like in this [issue](https://github.com/airbytehq/airbyte/issues/5724) 4-5 hours for 23 355 867 records. In case if instance cannot handle request (in my case it was not enough disk storage 50 GB instance for 14 GB table) it will fail the sync. Also to release instance memory I have to restart the instance.

The `setFetchSize(Integer.MIN_VALUE)` disables client-side caching of the entire response and gives you responses as they arrive there is no round-trip per row required. With this approach the sync starts immediately with any table size and no memory load on db. 

We tested sync between Mysql Source and S3 destination with 1 000 000 record table.
**Fetch size 10**
7 minutes to start the sync and 6 minutes to load 1 GB from table
**Fetch size  Integer.MIN_VALUE**
Start sync immediately and 6 minutes to load 1GB from table